runelite-client/src/main/java/net/runelite/client/plugins/poh/PohOverlay.java

```diff
@@ -0,0 +1,193 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.poh;
+
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+import javax.inject.Inject;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.runelite.client.plugins.poh.PohIcons.ALTAR;
+import static net.runelite.client.plugins.poh.PohIcons.ANNAKARL;
+import static net.runelite.client.plugins.poh.PohIcons.ARDOUGNE;
+import static net.runelite.client.plugins.poh.PohIcons.CAMELOT;
+import static net.runelite.client.plugins.poh.PohIcons.EXITPORTAL;
+import static net.runelite.client.plugins.poh.PohIcons.FALADOR;
+import static net.runelite.client.plugins.poh.PohIcons.FISHINGGUILD;
+import static net.runelite.client.plugins.poh.PohIcons.GLORY;
+import static net.runelite.client.plugins.poh.PohIcons.KHARYLL;
+import static net.runelite.client.plugins.poh.PohIcons.KOUREND;
+import static net.runelite.client.plugins.poh.PohIcons.LUMBRIDGE;
+import static net.runelite.client.plugins.poh.PohIcons.LUNARISLE;
+import static net.runelite.client.plugins.poh.PohIcons.MARIM;
+import static net.runelite.client.plugins.poh.PohIcons.POOLS;
+import static net.runelite.client.plugins.poh.PohIcons.REPAIR;
+import static net.runelite.client.plugins.poh.PohIcons.SENNTISTEN;
+import static net.runelite.client.plugins.poh.PohIcons.VARROCK;
+import static net.runelite.client.plugins.poh.PohIcons.WATERBIRTH;
+import static net.runelite.client.plugins.poh.PohIcons.YANILLE;
+
+public class PohOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2350;
+
+	@Getter
+	private final List<PohIcons> iconList = new ArrayList<>();
+
+	private final Client client;
+	private final PohConfig config;
+	private final PohPlugin plugin;
+
+	@Inject
+	public PohOverlay(Client client, PohConfig config, PohPlugin plugin)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
```

Comment:
After the default layer changes, layer needs to be set to ABOVE_WIDGETS here to render above resizeable minimap.